include/astshim/FitsChan.h

```diff
@@ -485,6 +487,14 @@ class FitsChan : public Channel {
     std::string getCardName() const { return getC("CardName"); }
 
     /**
+    Get the name of all cards, in order, starting from the first card
+
+    Not "const" because the code changes the index of the current card while operating
+    (but restores the original index when done).
+    */
+    std::vector<std::string> getCardNames();
```

Comment:
If this package is indeed bound for integration into AST and afw isn't exposing any of this, then I won't object to your naming choices. However, I do think it's worth considering `getAllCardNames` or something else that will distinguish it from `getCardName`.